tests/phpunit/tests/widgets.php

```diff
@@ -686,6 +686,61 @@ public function test_wp_widget_get_settings() {
 		$this->assertArrayNotHasKey( 0, $never_used );
 	}
 
+	/**
+	 * @ticket 54677
+	 *
+	 * @covers WP_Widget::get_settings
+	 */
+	public function test_wp_widget_initializes_widget_with_alt_option() {
+		/*
+		 * Emulate a new the recent posts widget.
+		 *
+		 * The widget contains an alternative (legacy) option so both the
+		 * current and the alternative option need to be deleted.
+		 */
+		delete_option( 'widget_recent-posts' );
+		delete_option( 'widget_recent_entries' );
+
+		$this->assertSame( false, get_option( 'widget_recent-posts' ), 'The option widget_recent-posts was not deleted.' );
+		$this->assertSame( false, get_option( 'widget_recent_entries' ), 'The option widget_recent_entries was not deleted.' );
+
+		wp_widgets_init();
+		$this->assertSameSetsWithIndex( array( '_multiwidget' => 1 ), get_option( 'widget_recent-posts' ), 'Option failed to be initialized.' );
+		$this->assertFalse( get_option( 'widget_recent_entries' ), 'Alternative option is set.' );
+	}
+
+	/**
+	 * @ticket 54677
+	 *
+	 * @covers WP_Widget::get_settings
+	 */
+	public function test_wp_widget_migrates_widget_with_alt_option() {
+		$option = array(
+			2              => array(
+				'title'     => 'Recent Posts',
+				'number'    => 5,
+				'show_date' => false,
+			),
+			'_multiwidget' => 1,
+		);
+
+		/*
+		 * Emulate the recent posts widget with an alternative option.
+		 *
+		 * The widget contains an alternative (legacy) option so the
+		 * current option is deleted while the alternative option is created.
+		 */
+		delete_option( 'widget_recent-posts' );
+		update_option( 'widget_recent_entries', $option );
+
+		$this->assertSame( false, get_option( 'widget_recent-posts' ), 'The option widget_recent-posts was not deleted.' );
+		$this->assertSameSets( $option, get_option( 'widget_recent_entries' ), 'The option widget_recent_entries was not set to the default.' );
```

Comment:
This uses `assertSameSets()`, but the later assertion uses `assertSameSetsWithIndex()`, as does the assertion in the previous test method.

Since `$option` is keyed, should this assertion also use `assertSameSetsWithIndex()`?